dask_planner/src/sql/logical/table_scan.rs

```diff
@@ -0,0 +1,54 @@
+use datafusion::logical_expr::LogicalPlan;
+use datafusion::logical_plan::TableScan;
+
+use crate::sql::exceptions::py_type_err;
+use pyo3::prelude::*;
+
+#[pyclass(name = "TableScan", module = "dask_planner", subclass)]
+#[derive(Clone)]
+pub struct PyTableScan {
+    pub(crate) table_scan: TableScan,
+}
+
+#[pymethods]
+impl PyTableScan {
+    #[pyo3(name = "getTableScanProjects")]
+    fn scan_projects(&mut self) -> PyResult<Vec<String>> {
+        let mut projs: Vec<String> = Vec::new();
+        match &self.table_scan.projection {
+            Some(e) => {
+                for index in e {
+                    projs.push(
+                        (&*self.table_scan.source.schema())
+                            .field(*index)
+                            .name()
+                            .clone(),
+                    );
+                }
+            }
+            None => (),
+        }
+        Ok(projs)
```

Comment:
We can avoid the mutable state and use functional code to map indices to field names. Also, this version just calls the table `schema()` method once, avoiding some unnecessary clones of Arc pointers, and removes the use of `&*`.

```suggestion
        match &self.table_scan.projection {
            Some(indices) => {
                let schema = self.table_scan.source.schema();
                Ok(indices
                    .iter()
                    .map(|i| schema.field(*i).name().to_string())
                    .collect())
            }
            None => Ok(vec![]),
        }
```

Reply:
love it. Will make the change